crates/api_crud/src/user/read.rs

```diff
@@ -96,13 +98,20 @@ impl PerformCrud for GetPersonDetails {
 
       // If its saved only, you don't care what creator it was
       // Or, if its not saved, then you only want it for that specific creator
-      if !saved_only.unwrap_or(false) {
-        posts_query = posts_query.creator_id(person_details_id);
-        comments_query = comments_query.creator_id(person_details_id);
-      }
-
-      let posts = posts_query.list()?;
-      let comments = comments_query.list()?;
+      let (posts, comments) = if !saved_only.unwrap_or(false) {
+        (
+          posts_query
+            .creator_id(Some(person_details_id))
+            .build()
+            .list()?,
+          comments_query
+            .creator_id(Some(person_details_id))
+            .build()
+            .list()?,
+        )
+      } else {
+        (posts_query.build().list()?, comments_query.build().list()?)
+      };
```

Comment:
This is pretty ugly but i dont see a better way, because the builder type changes when a setter is called. So the previous mut solution doesnt work.

Reply:
There is another builder macro which should support is, but disadvantage is that it doesnt get checked at compile time, and build() returns a result. Thats probably why this conditional isnt working.